### ISSUE TITLE
Cannot install package manager using corepack on node 16

### ISSUE THREAD

**Issue by SunsetTechuila** [actions/setup-node#1075]
**Description:**
No matter what node version you have chosen, the latest corepack will be used. The problem is that the latest corepack tries to use node's built-in fetch, which is not available in node 16
This is a [quite common](https://github.com/search?q=%22globalThis.fetch+is+not+a+function%22&type=pullrequests) problem

**Action version:**
v4

**Platform:**
- [x] Ubuntu
- [ ] macOS
- [ ] Windows

**Runner type:**
- [x] Hosted
- [ ] Self-hosted

**Tools version:**

Node: 16.20.2

**Repro steps:**  
1. Set the `packageManager` filed in the `package.json` file
2. Create a workflow containing these steps:

```yml
- uses: actions/checkout@v4

- run: corepack enable

- uses: actions/setup-node@v4
  with:
    node-version: 16
```

3. Run workflow

[workflow run](https://github.com/import-js/eslint-import-resolver-typescript/actions/runs/9327252035/job/25676880068)
[workflow file](https://github.com/import-js/eslint-import-resolver-typescript/blob/ca11f1c538714252ff058a8e1c680796ee5775d0/.github/workflows/ci.yml)
[example repo](https://github.com/SunsetTechuila/setup-node16-issue)

**Expected behavior:**
I expect the chosen package manager to be installed

**Actual behavior:**
```
Error: Error when performing the request to https://registry.npmjs.org/yarn/latest; 
  [cause]: TypeError: globalThis.fetch is not a function
      at fetch (/usr/local/lib/node_modules/corepack/dist/lib/corepack.cjs:22756:33)
```

**Comment by HarithaVattikuti** [comment]
Hello @SunsetTechuila 
Thank you for creating this issue. We will investigate it and get back to you as soon as we have some feedback.

**Comment by kuco23** [comment]
I am having the same issue with 

```
steps:
      - name: git checkout
        uses: actions/checkout@v4
      - name: Enable Corepack before setting up Node
        run: corepack enable
      - name: setup Node.js
        uses: actions/setup-node@v1
        with:
          node-version: 16.x
      - run: yarn
```
failing on `yarn`

**Comment by SunsetTechuila** [comment]
closing in favor of https://github.com/actions/runner-images/issues/10036

**Comment by SunsetTechuila** [comment]
update:

this happens because I run `corepack enable` before setting up Node

would be fixed by https://github.com/actions/setup-node/pull/901:

```yml
- name: Setup Node.js
  uses: JP250552/setup-node@feature/corepack
  with:
    node-version: 16
    corepack: true
```

**Comment by kuco23** [comment]
> update:
> 
> this happens because `corepack enable` is run before setting up Node
> 
> would be fixed by #901 since this works:
> 
> ```yaml
> - name: Setup Node.js
>   uses: JP250552/setup-node@feature/corepack
>   with:
>     node-version: 16
>     corepack: true
> ```
> 
> rn you can do this:
> 
> ```yaml
> - name: Setup target Node.js to enable Corepack
>   uses: actions/setup-node@v4
>   with:
>     node-version: 16
> 
> - name: Enable Corepack
>   run: corepack enable
> 
> - name: Setup Node.js
>   uses: actions/setup-node@v4
>   with:
>     node-version: 16
>     # or pnpm in case you don't use pnpm setup action
>     cache: yarn
> ```

confirm this fixed the issue
